eth/protocols/eth/peer.go

```diff
@@ -465,3 +424,35 @@ func (p *Peer) RequestTxs(hashes []common.Hash) error {
 		GetPooledTransactionsPacket: hashes,
 	})
 }
+
+type leakySet struct {
+	impl mapset.Set
```

Comment:
`impl` is really a no go. It doesn't mean anything. Please rename to `hashes`. also please add a method doc for the leakySet (renamed) type + constructor